native/packages/react-native-bpk-component-phone-input/src/BpkDiallingCodeList.ios.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import { View, StyleSheet, SectionList } from 'react-native';
+import React from 'react';
+import BpkText from 'react-native-bpk-component-text';
+import {
+  colorGray50,
+  colorGray100,
+  spacingSm,
+  spacingMd,
+} from 'bpk-tokens/tokens/base.react.native';
+import { BpkDiallingCodeListItem } from '..';
+import {
+  type ListCommonProps,
+  LIST_COMMON_PROP_TYPES,
+  LIST_COMMON_DEFAULT_PROPS,
+} from './common-types';
+
+const spacing = spacingMd * 1.5;
+
+const styles = StyleSheet.create({
+  sectionHeader: {
+    paddingVertical: spacingSm,
+    paddingHorizontal: spacing,
+    backgroundColor: colorGray50,
+  },
+  separator: {
+    flex: 1,
+    height: 1,
+    backgroundColor: colorGray100,
+    marginLeft: spacing,
+  },
+});
+
+/*
+Takes data in the format:
+[
+  {name: 'Foo'},
+  {name: 'Bar'},
+  {name: 'Baz'},
+]
+
+and returns:
+{
+  f: [{name: 'Foo'}],
+  b: [{name: 'Bar'}, {name: 'Baz'}],
+}
+*/
+const createAlphabetisedList = data => {
+  const list = {};
+  data.forEach(code => {
+    const firstLetter = code.name[0].toLowerCase();
+    if (list[firstLetter]) {
+      list[firstLetter].push(code);
+    } else {
+      list[firstLetter] = [code];
+    }
+  });
+  return list;
+};
+
+/*
+Takes data in the format:
+{
+  f: [{name: 'Foo'}],
+  b: [{name: 'Bar'}, {name: 'Baz'}],
+}
+
+and returns:
+[
+  {
+    title: 'F',
+    data: [{name: 'Foo'}],
+  },
+  {
+    title: 'B',
+    data: [{name: 'Bar'}, {name: 'Baz'}],
+  },
+]
+*/
+const convertCodesIntoSections = data => {
+  data.sort((a, b) => {
+    const nameA = a.name.toUpperCase();
+    const nameB = b.name.toUpperCase();
+    if (nameA < nameB) {
+      return -1;
+    }
+    if (nameA > nameB) {
+      return 1;
+    }
+    return 0;
+  });
+  const alphabetisedList = createAlphabetisedList(data);
+  return Object.keys(alphabetisedList).map(letter => ({
+    title: letter.toUpperCase(),
+    data: alphabetisedList[letter],
+  }));
+};
+
+type Props = {
+  ...$Exact<ListCommonProps>,
+};
+
+const BpkDiallingCodeList = ({ codes, selectedId, onItemPress }: Props) => (
+  <SectionList
+    sections={convertCodesIntoSections(codes)}
```

Comment:
I'd wanna look at the affects of this in render and a long list of say 196 entries.

Reply:
I've made a comment further down about performance.